docs/user_guide/02b_query_engine.md

```diff
@@ -1 +1,398 @@
 # Query Engine
+
+## What is the Query Engine?
+
+The **MedRecord Query Engine** enables users to find node and edges' indices stored in the graph structure efficiently. Thanks to an intuitive interface, complex queries can be performed, allowing you to filter of nodes and edges by their properties and relationships. This section introduces the basic concepts of querying MedRecords and explores advanced techniques for working with complex datasets.
+
+## Example dataset
+
+An example dataset can be used using the [`MedRecord`](medmodels.medrecord.medrecord.MedRecord){target="_blank"}'s method [`from_example_dataset`](medmodels.medrecord.medrecord.MedRecord.from_example_dataset){target="_blank"}.
+
+```{literalinclude} scripts/02b_show_dataset.py
+---
+language: python
+lines: 8
+---
+```
+
+This example dataset includes a set of patients, drugs, diagnoses and procedures. For this section, we will only use the patients, drugs and the edges that connect these two groups.
+
+```{exec-literalinclude} scripts/02b_show_dataset.py
+---
+language: python
+setup-lines: 1-42
+lines: 44
+---
+```
+
+```{exec-literalinclude} scripts/02b_show_dataset.py
+---
+language: python
+setup-lines: 1-42
+lines: 45
+---
+```
+
+```{exec-literalinclude} scripts/02b_show_dataset.py
+---
+language: python
+setup-lines: 1-42
+lines: 46
+---
+```
+
+## Node Queries
+
+The [`NodeOperand`](medmodels.medrecord.querying.NodeOperand){target="_blank"} querying class allow you to define specific criteria for selecting nodes within a [`MedRecord`](medmodels.medrecord.medrecord.MedRecord){target="_blank"}. These operands enable flexible and complex queries by combining multiple conditions, such as group membership, attributes' selection and querying, attribute values, and relationships to other nodes or edges. This section introduces the basic usage of node operands to create a powerful foundation for your data queries.
```

Comment:
I am not yet fully convinced about the naming of the queries. They are named: `query_node_basic `, `query_node_intermmediate`, `query_node_advanced`. Why that? From a users perspective, wouldn't it be more clean to read if the queries are named more like whats in it? So the query with the content `node.in_group("patient")` could be named for something that is more expressive to that content and the user is made aware that naming can be set to his or her choice.